examples/twittermap/web/public/javascripts/mapresultcache/services.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Module to cache map result data. Common services module communicates with
+ * middleware and updates resultcache module.
+ */
+'use strict';
+angular.module('cloudberry.mapresultcache', ['cloudberry.common'])
+  .service('MapResultCache', function () {
+
+    // The key-value store that stores mapResults
+    var store = new HashMap();
+    // To check if keyword in query changed
+    var currentKeywords = [""];
+    // To check if time range in query changed
+    var date = new Date();
+    date.setDate(date.getDate() - 1);
+    var currentTimeRange = {
+      start: new Date(),
+      end: date
+    };
+    // Prefix for geoIds to make key unique
+    var prefix = Object.freeze({
```

Comment:
could you teach me what does `freeze` mean? thanks!

Reply:
I used it like `enum`. Object.freeze() does not allow addition, modification or deletion of any property.